### PR TITLE
Output compilerOption Diagnostics

### PR DESCRIPTION
**Pull request by samcooke98**
## Rollup Plugin Name: `Typescript`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description

Typescript Rollup didn't output compiler options diagnostics. The most obvious case where this fails is when the supplied tsconfig leaves `composite: true`. (as declaration is ignored by rollup plugin) 

Previously, the plugin would fail silently, and print no detail about why it failed.

```
./src/index.ts → ./dist/index.js...
[!] (typescript plugin) Error: Couldn't compile 
src/index.ts
Error: Couldn't compile src/index.ts
    at Object.transform (/node_modules/@rollup/plugin-typescript/dist/index.js:466:23)
    at /node_modules/rollup/dist/rollup.js:16611:25

```

By adding the `services.getCompilerOptionsDiagnostics()` call the plugin will emit reasons for the failure.

### CODE REVIEW

File: packages/typescript/test/fixtures/invalid-compiler-options/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "composite": true,
+    "declaration": false,
+  },
+  "include": ["./"]
+}
```
Comment (NotWoods) [inline packages/typescript/test/fixtures/invalid-compiler-options/tsconfig.json]: Missing newline at the end

File: packages/typescript/test/test.js [hunk 2]
```diff
@@ -359,6 +359,29 @@ test('complies code that uses browser functions', async (t) => {
...
+        })
+      ],
+      onwarn({ toString, ...warning }) {
+        warnings.push(warning);
+      }
+    })
+  );
+
+  t.true(
+    err.message.includes('Composite projects may not disable declaration emit'),
```
Comment (NotWoods) [inline packages/typescript/test/test.js]: Should we override this flag too?

File: packages/typescript/test/test.js [hunk 3]
```diff
@@ -359,6 +359,29 @@ test('complies code that uses browser functions', async (t) => {
   t.true(code.includes('navigator.clipboard.readText()'), code);
 });
 
+test('reports compiler options errors', async (t) => {
+  process.chdir('fixtures/invalid-compiler-options');
+  const warnings = [];
```
Comment (NotWoods) [inline packages/typescript/test/test.js]: Since the warnings aren't used they should be removed

File: packages/typescript/test/test.js (hunk 2 above)
Comment (NotWoods) [reply packages/typescript/test/test.js]: I think we should add `composite` to our list of overridden compiler options if it causes an error.

Comment (samcooke98) [reply packages/typescript/test/test.js]: I agree - I think this is best done in a separate PR however. It seems ideal to output compilerOptions errors, regardless of the overridden compiler options - Future typescript versions may introduce new compilerOption errors. 

(And thinking about it, when I do this in a separate PR, it'll break this test)

**Comment by shellscape** [comment]
@samcooke98 apologies that this one sat so long. Any chance you'd be willing to address the conflicts and failing CI?

**Comment by samcooke98** [comment]
> @samcooke98 apologies that this one sat so long. Any chance you'd be willing to address the conflicts and failing CI?

No worries, it's my bad too, it slipped off my radar. I'll try to get to it this weekend (ie within the next 3 days)

**Comment by shellscape** [comment]
@samcooke98 we had some CI shenanigans going that were fixed up today. Please do a merge from master and push and it should start passing.

**Comment by samcooke98** [comment]
Just tested this with`@rollup/plugin-typescript@4.1.1` and couldn't reproduce - seems fixed? 

Will close now (Sorry I forgot about this PR)

**Pull request closed without merging**